doc/designs/passkeys.md

```diff
@@ -0,0 +1,246 @@
+`ipa user-add-passkey LOGIN --register`.
+
+
+### Access control
+
+#### Permissions
+
+- New permission created for writing the passkey configuration:
+`System: Modify Passkey Configuration`. Granted to the Privilege `Passkey Administrators`
+- New permission created for reading the passkey configuration:
+`System: Read Passkey configuration`. Granted to all authenticated users.
+- New permission for managing passkey mapping:
+`System: Manage Passkey Mappings`. Granted to the Privilege: `Passkey Administrators`
+
+- Extend existing permission" `System: Read User Addressbook Attributes`:
+allow read access to the ipapasskey attribute (granted to all authenticated
+users). This attribute is not sensitive as it contains only public data.
+
+#### Self-service Permission
+
+- New self-service permission for managing their own passkey mapping:
+`Users can manage their own passkey mappings`
+
+#### Privilege
+
+- New privilege `Passkey Administrators` with the permissions `System: Modify Passkey Configuration` and `System: Manage Passkey Mappings`.
+
+By default only members of the admins group are allowed to modify the passkey
+settings or another user's passkeys.
+
+## Implementation
+
+### LDAP schema
+
+New objectclass and attribute for the passkey configuration object:
+```
+attributeTypes: ( 2.16.840.1.113730.3.8.24.26 NAME 'ipaRequireUserVerification' DESC 'require passkey user verification' SYNTAX 1.3.6.1.4.1.1466.115.121.1.15 SINGLE-VALUE X-ORIGIN 'IPA v4.10')
+objectclasses: ( 2.16.840.1.113730.3.8.24.8 NAME 'ipaPasskeyConfigObject' DESC 'IPA passkey global config options' AUXILIARY MUST ipaRequireUserVerification X-ORIGIN 'IPA v4.10')
+```
+
+New objectclass and attribute for the passkey mapping:
+```
+attributeTypes: ( 2.16.840.1.113730.3.8.24.27 NAME 'ipapasskey' DESC 'Passkey mapping' EQUALITY caseExactMatch SYNTAX 1.3.6.1.4.1.1466.115.121.1.15 X-ORIGIN 'IPA v4.10' )
+objectclasses: ( 2.16.840.1.113730.3.8.24.9 NAME 'ipaPasskeyUser' DESC 'IPA passkey user' AUXILIARY MAY ipapasskey X-ORIGIN 'IPA v4.10')
+```
+
+### Indices
+
+No need to add a new index for ipapasskey as the search performed by SSSD
+will use a filter based on the user uid.
+
+
+## Feature Management
+
+### UI
+
+- A new tab will be added below "Policy", at the same level as `Host-Based Access Control`, `Sudo`, `SELInux User Maps`, `Password Policies` and `Kerberos Ticket Policy`, with the label `Passkey Configuration`.
+
+It will allow to configure the attribute `Require User Verification`, with a radio button: `on`, `off` or `default`.
+
+- In the `User` facet, a new field will be added, below `SSH public keys`, with the label `Passkey mappings`, and will display the values, or allow to add a new value.
+
+Note: since the Web browser my be running on a non-enrolled host without
+the required packages, the WebUI will probably need specific javascript code
+to register a key by inserting it on the machine where the browser is
+running.
```

Comment:
I think we need to move the whole registration and login process into a separate app running on IPA server. I have plans for this for OAuth2 integration already. For the purpose of this design you may add that this process not be part of the original implementation.